front/src/app/components/forms/register-agencies/RegisterAgenciesForm.tsx

```diff
@@ -44,42 +48,52 @@ export const RegisterAgenciesForm = ({
       }),
     );
   };
+
+  useEffect(() => {
+    if (inputValue === undefined && initialSiret) {
+      setInputValue(initialSiret);
+    }
+  }, [inputValue, initialSiret]);
+
   return (
     <>
-      <div className={fr.cx("fr-mt-4w", "fr-mb-2w")}>
-        <Input
-          label="Rechercher un organisme"
-          hintText="Par n° SIRET ou nom sous lequel il est enregistré sur Immersion Facilitée"
-          nativeInputProps={{
-            id: domElementIds.agencyDashboard.registerAgencies.search,
-            type: "search",
-            placeholder: "",
-            value: inputValue,
-            onChange: onAgencySearchChange,
-            onKeyDown: (event) => {
-              if (event.key === "Escape") {
-                agencySearchBySiretOrNameInput.current?.blur();
-              }
-            },
-          }}
-        />
-      </div>
-      <a
-        href="https://annuaire-entreprises.data.gouv.fr/"
-        target="_blank"
-        rel="noreferrer"
-        className={fr.cx("fr-link")}
-      >
-        <i className={fr.cx("fr-icon-information-fill", "fr-icon--sm")} />
-        Retrouver votre siret sur l'Annuaire des Entreprises
-      </a>
+      <>
+        <div className={fr.cx("fr-mt-4w", "fr-mb-2w")}>
+          <Input
+            label="Se rattacher à un organisme"
+            hintText="Rechercher par n° SIRET ou nom sous lequel il est enregistré sur Immersion Facilitée"
+            nativeInputProps={{
+              id: domElementIds.agencyDashboard.registerAgencies.search,
+              type: "search",
+              placeholder: "",
+              value: inputValue,
+              onChange: onAgencySearchChange,
+              onKeyDown: (event) => {
+                if (event.key === "Escape") {
+                  agencySearchBySiretOrNameInput.current?.blur();
+                }
+              },
+            }}
+          />
+        </div>
+        <a
+          href="https://annuaire-entreprises.data.gouv.fr/"
+          target="_blank"
+          rel="noreferrer"
+          className={fr.cx("fr-link")}
+        >
+          <i className={fr.cx("fr-icon-information-fill", "fr-icon--sm")} />
+          Retrouver votre siret sur l'Annuaire des Entreprises
+        </a>
+      </>
 
-      {inputValue.length > 0 && (
+      {inputValue && inputValue.length > 0 && (
```

Comment:
```suggestion
      {inputValue && (
```
Je pense que c'est plus nécessaire (sauf si on veut faire ceinture et bretelles)